decisions/DR004-Core-language-support.md

```diff
@@ -0,0 +1,476 @@
+    typedef struct ns(StringMapOpaque) * ns(StringMapHandle);
+
+    // Create
+    ns(StringMapHandle) ns(StringMap_create)()
+    {
+        using openassetio::CHandle;
+        using openassetio::StringMap;
+        return CHandle::create<StringMap, ns(StringMapHandle)>();
+    }
+
+    // Release
+    void ns(StringMap_release)(ns(StringMapHandle) handle)
+    {
+        using openassetio::CHandle;
+        using openassetio::StringMap;
+        CHandle::release<StringMap>(handle);
+    }
+
+    // Setter
+    void ns(StringMap_set)(ns(StringMapHandle) handle, const char * key, const char * val)
+    {
+        using openassetio::CHandle;
+        using openassetio::StringMap;
+        auto & ptrStringMap = CHandle::to_cpp<StringMap>(handle);
+        ptrStringMap->insert({key, val});
+    }
+}
+
+////////////////////////////////////////////////////////////////////////
+// Dummy Manager declaration
+
+#undef ns
+#define ns(thing) OpenAssetIO_C_NAMESPACE(hostAPI_##thing)
+
+namespace openassetio
+{
+inline namespace OpenAssetIO_VERSION
+{
+namespace hostAPI
+{
+struct Manager
+{
+    void updateTerminology(StringMap & terminology) const;
+    std::string displayName() const;
+};
+}  // namespace hostAPI
+}  // namespace OpenAssetIO_VERSION
+}  // namespace openassetio
+extern "C"
+{
+    typedef struct ns(ManagerOpaque) * ns(ManagerHandle);
+}
+
+////////////////////////////////////////////////////////////////////////
+// Default terminology
+
+#undef ns
+#define ns(thing) OpenAssetIO_C_NAMESPACE(terminology_##thing)
+
+extern "C"
+{
+    [[maybe_unused]] static const char * ns(kTerm_Asset) = "asset";
+    [[maybe_unused]] static const char * ns(kTerm_Assets) = "assets";
+    [[maybe_unused]] static const char * ns(kTerm_Manager) = "manager";
+    [[maybe_unused]] static const char * ns(kTerm_Publish) = "publish";
+    [[maybe_unused]] static const char * ns(kTerm_Publishing) = "publishing";
+    [[maybe_unused]] static const char * ns(kTerm_Published) = "published";
+    [[maybe_unused]] static const char * ns(kTerm_Shot) = "shot";
+    [[maybe_unused]] static const char * ns(kTerm_Shots) = "shots";
+}
+namespace openassetio
+{
+inline namespace OpenAssetIO_VERSION
+{
+namespace terminology
+{
+const StringMap defaultTerminology{
+    {ns(kTerm_Asset), "entity"}, {ns(kTerm_Published), "written"},
+    // ... etc
+};
+}
+}  // namespace OpenAssetIO_VERSION
+}  // namespace openassetio
+extern "C"
+{
+    // Getter
+    const char * ns(get_default_term)(const char * key)
+    {
+        // Static, so safe to return raw pointer.
+        return openassetio::terminology::defaultTerminology.at(key).c_str();
+    }
+}
+////////////////////////////////////////////////////////////////////////
+// Mapper class
+
+namespace openassetio
+{
+inline namespace OpenAssetIO_VERSION
+{
+
+namespace terminology
+{
+
+std::string replaceTerms(const StringMap & terminology, const std::string & sourceStr)
```

Comment:
so like here. I'm sure there's a notion of a well formed terminology string, and a malformed one.